Chapter1/4_Palindrome Permutation/PalindromePermutation.py

```diff
@@ -2,20 +2,14 @@
 import unittest
 
 
-def pal_perm(phrase):
-    '''function checks if a string is a permutation of a palindrome or not'''
-    table = [0 for _ in range(ord('z') - ord('a') + 1)]
-    countodd = 0
-    for c in phrase:
-        x = char_number(c)
-        if x != -1:
-            table[x] += 1
-            if table[x] % 2:
-                countodd += 1
-            else:
-                countodd -= 1
-
-    return countodd <= 1
+def palindrome_permutation(string):
+    # No lowercase, ignore special characters
+    string = [char for char in string.lower() if ord(char) in range(97, 123)]
+    # to be a palindrome, the count of each letter has to be even, save for one letter that can be odd.
+    my_dict = {char: string.count(char) for char in set(string)}
```

Comment:
the big-O here appears to be roughly N^2 : `string_length * number_of_unique_characters`